File: react-native-app/src/components/StatusBadge.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import Colors from '../theme/colors';

const BADGE_CONFIG = {
  done: { label: 'Done', icon: 'checkmark-circle', fg: Colors.success, bg: Colors.successBg },
  missed: { label: 'Missed', icon: 'close-circle', fg: Colors.emergency, bg: Colors.emergencyBg },
  sent: { label: 'Sent', icon: 'arrow-up-circle', fg: Colors.primaryAction, bg: Colors.blueBg },
  todo: { label: 'To Do', icon: 'ellipse-outline', fg: Colors.mutedText, bg: Colors.subtleBg },
};

export default function StatusBadge({ type }) {
  const config = BADGE_CONFIG[type] || BADGE_CONFIG.todo;

  return (
    <View
      style={[styles.container, { backgroundColor: config.bg }]}
      accessibilityLabel={`Status: ${config.label}`}
    >
      <Ionicons name={config.icon} size={16} color={config.fg} />
      <Text style={[styles.label, { color: config.fg }]}>{config.label}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 10,
    paddingVertical: 4,
    borderRadius: 12,
    gap: 4,
  },
  label: {
    fontSize: 16,
    fontWeight: '700',
  },
});

File: flutter_app/lib/widgets/status_badge.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';

enum BadgeType { done, missed, sent, todo }

class StatusBadge extends StatelessWidget {
  final BadgeType type;

  const StatusBadge({
    super.key,
    required this.type,
  });

  @override
  Widget build(BuildContext context) {
    final (label, icon, fg, bg) = switch (type) {
      BadgeType.done => ('Done', Icons.check_circle_outline, AppColors.success, AppColors.successBg),
      BadgeType.missed => ('Missed', Icons.error_outline, AppColors.warningDark, AppColors.warningBg),
      BadgeType.sent => ('Sent', Icons.send, AppColors.primaryAction, AppColors.blueBg),
      BadgeType.todo => ('To Do', Icons.circle_outlined, AppColors.heading, AppColors.blueBg),
    };

    return Container(
      constraints: const BoxConstraints(minWidth: 48, minHeight: 48),
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
      decoration: BoxDecoration(
        color: bg,
        borderRadius: BorderRadius.circular(20),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          ExcludeSemantics(child: Icon(icon, size: 18, color: fg)),
          const SizedBox(width: 6),
          Text(
            label,
            style: TextStyle(
              fontSize: 16,
              fontWeight: FontWeight.w700,
              color: fg,
            ),
          ),
        ],
      ),
    );
  }
}
